testsuites/sparql/src/main/java/org/eclipse/rdf4j/testsuite/query/parser/sparql/manifest/SPARQLComplianceTest.java

```diff
@@ -67,18 +67,14 @@ public DynamicSparqlComplianceTest(String displayName, String testURI, String na
 			this.name = name;
 		}
 
-		public void test() {
+		public void test() throws Exception {
 			assumeThat(getIgnoredTests().contains(getName())).withFailMessage("test case '%s' is ignored", getName())
 					.isFalse();
 			try {
 				setUp();
 				runTest();
-			} catch (Exception e) {
-				try {
-					tearDown();
-				} catch (Exception e2) {
-
-				}
+			} finally {
+				tearDown();
 			}
```

Comment:
This code was swallowing exceptions and also the tearDown() method was only called if there was an exception.